packages/adapters/storage/src/postgres.rs

```diff
@@ -1252,37 +1223,62 @@ impl Postgres {
     }
 }
 
-async fn take_blocks_until_limit(
-    mut stream: BoxStream<'_, std::result::Result<tables::DBCompressedFuelBlock, sqlx::Error>>,
-    max_cumulative_bytes: u32,
-) -> Result<Vec<CompressedFuelBlock>> {
-    let mut blocks = vec![];
+// Helper function to count additional blocks until the buildup threshold is reached or the stream is exhausted.
+async fn count_remaining_blocks(
```

Comment:
If you limit the number of results in the query, then you don't need this function. 

Because instead of `total_blocks_count >= block_buildup_threshold` you can use `total_blocks_count == block_buildup_threshold` to detect `buildup_detected`

Reply:
I cannot limit the number of blocks in the query because, even though there might be a buildup, the blocks could be very small and I could still consume them all and not go over the max cumulative byte limit.

E.g.

We have 36k unbundled blocks taking up a total of 1MB in the DB. Anything over 3600 is considered a buildup. If I limit the query to 3.6k rows I'm only going to get 3.6k blocks (100kB) while I could have gotten all 36k (1MB) because they are so small.

Reply:
Okay, I see, the first loop can process all 36k entries before it touches the buildup logic.